File: new-devcenter/replace_code_blocks.js
const fs = require('fs');

// Read the file
const content = fs.readFileSync('src/pages/DevCenter.tsx', 'utf-8');

// Pattern to match <pre className="bg-accent..."><code>{`...`}</code></pre>
const pattern = /<pre className="bg-accent p-4 rounded-lg overflow-x-auto(?:\s+text-sm)?(?:\s+mb-\d+)?">\s*<code>(\{`[\s\S]*?`\})<\/code>\s*<\/pre>/g;

let count = 0;
const newContent = content.replace(pattern, (match, codeContent) => {
    count++;
    // Extract mb-X class if exists
    const mbMatch = match.match(/mb-(\d+)/);
    const mbClass = mbMatch ? `\n                className="mb-${mbMatch[1]}"` : '';
    
    return `<CodeBlock 
                language="kotlin"
                code=${codeContent}${mbClass}
              />`;
});

// Write back
fs.writeFileSync('src/pages/DevCenter.tsx', newContent, 'utf-8');

console.log(`Successfully replaced ${count} code blocks!`);

File: new-devcenter/replace_code_blocks.py
import re

# Read the file
with open('src/pages/DevCenter.tsx', 'r', encoding='utf-8') as f:
    content = f.read()

# Pattern to match <pre className="bg-accent..."><code>{`...`}</code></pre>
pattern = r'<pre className="bg-accent p-4 rounded-lg overflow-x-auto(?:\s+text-sm)?(?:\s+mb-\d+)?">\s*<code>(\{`(?:[^`]|`(?!\}))*`\})</code>\s*</pre>'

def replace_pre_code(match):
    code_content = match.group(1)
    # Extract className modifiers
    full_match = match.group(0)
    mb_match = re.search(r'mb-(\d+)', full_match)
    mb_class = f'\n                className="mb-{mb_match.group(1)}"' if mb_match else ''
    
    return f'''<CodeBlock 
                language="kotlin"
                code={code_content}{mb_class}
              />'''

# Replace all occurrences
new_content = re.sub(pattern, replace_pre_code, content, flags=re.DOTALL)

# Write back
with open('src/pages/DevCenter.tsx', 'w', encoding='utf-8') as f:
    f.write(new_content)

print(f'Successfully replaced code blocks!')
